### ISSUE TITLE
Bots - Doesn't fully climb ladder and onto platforms before interacting with objects (Repairs/Operate)

### ISSUE THREAD

**Issue by Kaillera** [Regalis11/Barotrauma#3707]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Bots doesn't fully climb ladder and onto platforms before interacting with objects (Repairs/Operate) that are positioned next to ladders on platform. They tend to climb most of the way up a ladder and if they can interact with the object, they'll try. Usually this ends up with them falling down the platform and giving up climbing the ladder again.

**Steps To Reproduce**
Good example is [Kail Terra](https://steamcommunity.com/sharedfiles/filedetails/?id=2197390262)'s engine room's periscope.

![Capture](https://user-images.githubusercontent.com/5086682/92229811-0e340d80-ee78-11ea-8edc-66985449e2bd.PNG)

It's about 50/50 whether or not a bot would get onto the platform first before they interact with the periscope.

**Version**
~0.10.501

**Comment by itchyOwl** [comment]
It wasn't easy to reproduce, but I managed to get the issue appear when I had the operate order issued and teleported the character around so that it ran back to the periscope. It took something like 10 attempts to see the issue.

Nevertheless, I think you are right. There were not enough checks for the climbing, causing the bot to interact with the periscope too early. This in turn caused it to release the ladders and to drop through the platform.

Added a bunch of checks in Regalis11/Barotrauma-development@ca073fb (human-ai), after which I wasn't able to reproduce the issue anymore. This fix is not included in the next update, but give it a test later.

Thanks for reporting. Also found another pathing related issue while testing the issue 👍
